client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -0,0 +1,351 @@
+		self.label_file = self.builder.get_object('SFTPClientGUI.label')
+		col = Gtk.TreeViewColumn('Transfer')
+		colText = Gtk.CellRendererText()
+		col.pack_start(colText, True)
+		col.add_attribute(colText, "text", 0)
+		self.treeview_transfer.append_column(col)
+		self._tvmodel = Gtk.ListStore(str)
+		self.treeview_transfer.connect('size-allocate', self._treeview_changed)
+		self.treeview_transfer.set_model(self._tvmodel)
+		self.treeview_transfer.connect('button_press_event', self._signal_treeview_button_pressed)
+
+		self.popup_menu = Gtk.Menu.new()
+		menu_item = Gtk.MenuItem.new_with_label('Clear')
+		menu_item.connect('activate', self.clear_log)
+		self.popup_menu.append(menu_item)
+		self.popup_menu.show_all()
+
+	def _signal_treeview_button_pressed(self, _, event):
+		if event.button == 3:
+			self.popup_menu.popup(None, None, None, None, event.button, Gtk.get_current_event_time())
+			return True
+		return
+
+	def log_generic(self, string):
+		self._tvmodel.prepend((string,))
+
+	def clear_log(self, _):
+		self._tvmodel.clear()
+
+	def _treeview_changed(self, widget, event, data=None):
+		adj = self.scroll.get_vadjustment()
+		adj.set_value(0)
+
+	def log_transfer_upload(self, path, size, isFolder, error=False):
+		if error:
+			string = 'Error uploading ' + path
+			self._tvmodel.prepend((string,))
+		elif isFolder:
+			string = 'Uploading folder ' + path + ' and all its children...'
+			self._tvmodel.prepend((string,))
+		else:
+			string = 'Uploading ' + path
+			self._tvmodel.prepend((string,))
+
+class DirectoryBase(object):
+	def __init__(self, treeview, logger):
+		col = Gtk.TreeViewColumn("Files")
+		self.logger = logger
+		self.treeview_local = treeview
+		colText = Gtk.CellRendererText()
+		colImg = Gtk.CellRendererPixbuf()
+		col.pack_start(colImg, False)
+		col.pack_start(colText, True)
+		col.add_attribute(colText, "text", 0)
+		col.add_attribute(colImg, "pixbuf", 1)
+		col.set_sort_column_id(0)
+		col_perm = Gtk.TreeViewColumn("Permissions")
+		col_perm.pack_start(colText, True)
+		col_perm.add_attribute(colText, "text", 3)
+		col_size = Gtk.TreeViewColumn("Size")
+		col_size.pack_start(colText, True)
+		col_size.add_attribute(colText, "text", 4)
+		col_size.set_sort_column_id(5)
+		col_date = Gtk.TreeViewColumn("Date Modified")
+		col_date.pack_start(colText, True)
+		col_date.add_attribute(colText, "text", 6)
+		col_date.set_sort_column_id(6)
+		self.treeview_local.append_column(col)
+		self.treeview_local.append_column(col_perm)
+		self.treeview_local.append_column(col_size)
+		self.treeview_local.append_column(col_date)
+
+		self.treeview_local.connect('row-expanded', self.expand_row)
+		self.treeview_local.connect('row-collapsed', self.collapse_row)
+		self._tvmodel = Gtk.TreeStore(str, Pixbuf, str, str, str, int, str)
+
+		self.local_hidden = True
+		self.treeview_local.connect('button_press_event', self._signal_treeview_button_pressed)
+
+		self.popup_menu = Gtk.Menu.new()
```

Comment:
This `__init__` method is already rather large, moving the menu generation to a `_get_popup_menu()` method would be helpful. Then just set `self.popup_menu = self._get_popup_menu()`.